models/DataObject/AbstractObject.php

```diff
@@ -242,6 +243,11 @@ public static function getById(int|string $id, array $params = []): ?static
                     $object = self::getModelFactory()->build($className);
                     RuntimeCache::set($cacheKey, $object);
                     $object->getDao()->getById($id);
+                    // fire pre load event
+                    $preLoadEvent = new DataObjectPreLoadEvent($object, ['params' => $params]);
+                    \Pimcore::getEventDispatcher()->dispatch($preLoadEvent, DataObjectEvents::PRE_LOAD);
+                    $object = $preLoadEvent->getObject();
```

Comment:
Is it really a *pre*-load event if it gets dispatched *after* the element was loaded from the database? Any why is `pimcore.*.postLoad` not enough?